public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx

```diff
@@ -103,15 +91,86 @@ export class PanelHeaderCorner extends Component<Props> {
     }
 
     if (infoMode === InfoMode.Error && error) {
-      return this.renderCornerType(infoMode, error, this.onClickError);
+      return <PanelInfoCorner infoMode={infoMode} content={error} onClick={this.onClickError} />;
     }
 
     if (infoMode === InfoMode.Info || infoMode === InfoMode.Links) {
-      return this.renderCornerType(infoMode, this.getInfoContent);
+      return <PanelInfoCorner infoMode={infoMode} content={this.getInfoContent} />;
     }
 
     return null;
   }
 }
 
 export default PanelHeaderCorner;
+
+interface PanelInfoCornerProps {
+  infoMode: InfoMode;
+  content: PopoverContent;
+  onClick?: () => void;
+}
+
+function PanelInfoCorner({ infoMode, content, onClick }: PanelInfoCornerProps) {
+  const theme = infoMode === InfoMode.Error ? 'error' : 'info';
+  const ariaLabel = selectors.components.Panels.Panel.headerCornerInfo(infoMode.toLowerCase());
+  const styles = useStyles2(getStyles);
+
+  return (
+    <Tooltip content={content} placement="top-start" theme={theme} interactive>
+      <button type="button" className={styles.infoCorner} onClick={onClick} aria-label={ariaLabel}>
+        <Icon
+          name={iconMap[infoMode]}
+          size={infoMode === InfoMode.Links ? 'sm' : 'lg'}
+          className={cx(styles.icon, infoMode === InfoMode.Links ? styles.iconLinks : '')}
```

Comment:
proper way to conditionally apply styles using `cx` would be something like:
```suggestion
          className={cx(styles.icon, {
            [styles.iconLinks]: infoMode === InfoMode.Links
          })}
```

same applies a couple of lines below